### PR TITLE
local docker test environment

### PR DESCRIPTION
**Pull request by krisss85**
The purpose of this PR is to setup local docker test environment

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+version: '3'
+
+volumes:
+  mysql_data:
+      driver: local
+  realm-export.json:
+
+services:
+  mysql:
+      image: mysql:5.7
```
Comment (bolkedebruin) [inline docker/docker-compose.yml:10]: Are you doing this for persistence? I can run keycloak (quay.io/keycloak/keycloak , which is the recommended one: https://www.keycloak.org/getting-started/getting-started-docker) without it

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      volumes:
+        - ${PWD}/realm-export.json:/export/realm-export.json
+      environment:
+        DB_VENDOR: MYSQL
+        DB_ADDR: mysql
+        DB_DATABASE: keycloak
+        DB_USER: keycloak
+        DB_PASSWORD: password
+        KEYCLOAK_USER: admin
+        KEYCLOAK_PASSWORD: Pa55w0rd
```
Comment (bolkedebruin) [inline docker/docker-compose.yml:31]: I suggest just using admin, as this is for development and no one will remember this one after a while ;-)

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      ports:
+        - 9443:9443
+      restart: on-failure
+      depends_on:
+        - keycloak
+      healthcheck:
+        test: ["CMD", "curl", "-f", "http://keycloak:8080"]
+        interval: 30s
+        timeout: 10s
+        retries: 10
```
Comment (bolkedebruin) [inline docker/docker-compose.yml:50]: add a new line plz

File: docker/rdpgw.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ providerUrl: http://keycloak:8080/auth/realms/master
+ clientId: rdpgw
+ clientSecret: 9defc114-616d-40b3-a9cd-661488f09e5e
+caps:
+ tokenAuth: true
+ idleTimeout: 10
+ DisablePrinter: true
+ DisablePort: true
+ DisablePnp: true
+ DisableDrive: true
```
Comment (bolkedebruin) [inline docker/rdpgw.yaml:30]: new line please

File: docker/docker-compose.yml (hunk 1 above)
Comment (krisss85) [reply docker/docker-compose.yml:10]: But then you face a problem that you have to restart the container
`keycloak_1  | Added 'admin' to '/opt/jboss/keycloak/standalone/configuration/keycloak-add-user.json', restart server to load user
keycloak_1  | -b 0.0.0.0`

Comment (bolkedebruin) [reply docker/docker-compose.yml:10]: Check, its just nice to have it a bit lighter.

Comment (krisss85) [reply docker/docker-compose.yml]: sure thing, it should be addressed now, i created a separate realm for rdpgw

File: docker/docker-compose.yml (hunk 2 above)
Comment (krisss85) [reply docker/docker-compose.yml]: :+1:

File: docker/docker-compose.yml (hunk 3 above)
Comment (krisss85) [reply docker/docker-compose.yml]: all good now?

File: docker/rdpgw.yaml (hunk 4 above)
Comment (krisss85) [reply docker/rdpgw.yaml]: all good now?

**Comment by krisss85** [comment]
@bolkedebruin i have addressed all the comments, it is now fully fledged self-contained testing env, please try it out according to readme and let me know if there is anything missing still

**Comment by bolkedebruin** [comment]
Awesome looks good!

**Pull request merged**